app/support/auto_expire_reservation.rb

```diff
@@ -15,7 +15,8 @@ def order_details
 
   def purchased_active_order_details
     OrderDetail.purchased_active_reservations
-      .merge(Reservation.relay_in_progress)
+      .joins("INNER JOIN products ON order_details.product_id = products.id")
+      .joins("INNER JOIN relays ON relays.instrument_id = products.id")
```

Comment:
No unfortunately. I tried `.joins(:product).joins(:relay)` and `.joins(product: :relay)`, but it's complaining there's no relay association on product.

```
ActiveRecord::ConfigurationError:
       Association named 'relay' was not found on Product; perhaps you misspelled it?
     # ./app/support/auto_expire_reservation.rb:21:in `purchased_active_order_details'
     # ./app/support/auto_expire_reservation.rb:13:in `order_details'
     # ./app/support/auto_expire_reservation.rb:3:in `perform'
     # ./spec/app_support/auto_expire_reservation_spec.rb:145:in `block (5 levels) in <top (required)>'
     # ./spec/app_support/auto_expire_reservation_spec.rb:145:in `block (4 levels) in <top (required)>'
```